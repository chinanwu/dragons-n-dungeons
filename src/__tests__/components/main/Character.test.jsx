import { shallow } from 'enzyme';
import React from 'react';

import {
  Character,
  mapStateToProps,
} from '../../../components/main/Character.jsx';

jest.unmock('../../../components/main/Character.jsx');

describe('Character component', () => {
  describe('rendering', () => {
    it('renders Character', () => {
      const wrapper = shallow(<Character />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('changes detail', () => {
      const onChange = jest.fn();
      const event = {
        target: {
          value: 'test2',
        },
      };
      const wrapper = shallow(<Character name="test" onChange={onChange} />);
      expect(wrapper.getElement()).toMatchSnapshot();
      wrapper.find('#characterName').simulate('change', event);
      expect(onChange.mock.calls).toEqual([['name', 'test2']]);
    });
  });

  describe('mapStateToProps', () => {
    it('returned mapped properties', () => {
      const state = {
        name: 'name',
        level: 'level',
        race: 'race',
        cClass: 'cClass',
        experience: 'experience',
        alignment: 'alignment',
        background: 'background',
      };
      expect(mapStateToProps({ character: state })).toEqual(state);
    });
  });
});
