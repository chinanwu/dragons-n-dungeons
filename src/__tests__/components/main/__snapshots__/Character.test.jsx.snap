// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component behaviour changes detail 1`] = `
<div
  className="Character"
>
  <div
    className="Character__name"
  >
    <LabelledInput
      id="characterName"
      label="Name: "
      onChange={[Function]}
      type="string"
      value="test"
    />
  </div>
  <div
    className="Character--bottom"
  >
    <div
      className="Character--left"
    >
      <LabelledInput
        id="characterLevel"
        label="Level: "
        onChange={[Function]}
        type="number"
      />
      <LabelledInput
        id="characterRace"
        label="Race: "
        onChange={[Function]}
        type="text"
      />
      <LabelledInput
        id="characterClass"
        label="Class: "
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="Character--right"
    >
      <LabelledInput
        id="characterExperience"
        label="Experience: "
        onChange={[Function]}
        type="number"
      />
      <LabelledInput
        id="characterAlignment"
        label="Alignment: "
        onChange={[Function]}
        type="text"
      />
      <LabelledInput
        id="characterBackground"
        label="Background: "
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`Character component rendering renders Character 1`] = `
<div
  className="Character"
>
  <div
    className="Character__name"
  >
    <LabelledInput
      id="characterName"
      label="Name: "
      onChange={[Function]}
      type="string"
    />
  </div>
  <div
    className="Character--bottom"
  >
    <div
      className="Character--left"
    >
      <LabelledInput
        id="characterLevel"
        label="Level: "
        onChange={[Function]}
        type="number"
      />
      <LabelledInput
        id="characterRace"
        label="Race: "
        onChange={[Function]}
        type="text"
      />
      <LabelledInput
        id="characterClass"
        label="Class: "
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="Character--right"
    >
      <LabelledInput
        id="characterExperience"
        label="Experience: "
        onChange={[Function]}
        type="number"
      />
      <LabelledInput
        id="characterAlignment"
        label="Alignment: "
        onChange={[Function]}
        type="text"
      />
      <LabelledInput
        id="characterBackground"
        label="Background: "
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;
